control/control.go

```diff
@@ -114,23 +114,24 @@ func (c *Control) Run() {
 				}
 			}
 
-			if Q > 0 {
-				// Q > 0 means the system is queued up so just use a y-dot
-				// estimation of the rate since workers are operating at full
-				// speed.
+			if Q > B {
```

Comment:
If `R` is sufficiently high and `DX` and `DY` are much bigger than zero, does this give us a very small window?

For example, yesterday we were at `B=5`, `DX=500`, `DY=500`, and `R=100`. That means that `Q=6` would be our threshold here.

I feel like there's still going to be friction here at very low `B` values. For very efficient workers, this window will be more narrow. Intuitively, I'd expect this window to be broader the more efficient and high volume the workload is. This probably is Fine ™️ in practice, as going from 4->3 workers isn't super important and going from 3->2 workers is probably forbidden by our MinCount settings.

Reply:
(I don't think this is a blocker to rolling this out, just poking on some of the elegance here)

Reply:
With the process we were testing, Q kept hovering between 0 and around 5 or less. Note that we don't need it to be under this limit too often either -- just once every some control periods, and it shall be fine. At very low values, either you're on the correct setting (b = 1? no need in going lower than that. b = 2? double the chance already.)

>For very efficient workers, this window will be more narrow.
Wouldn't very efficient workers pick up messages real fast, making Q very low?

Reply:
I think, as you put it, that even if for some cases the result isn't ideal, it's Fine :tm: in practice given that the system will self-correct if it pushes too much towards an end. We just need it to push.